### DISCUSSION TITLE
Moving to Sway

### DISCUSSION THREAD

**Discussion by Jarvvski** [regolith-linux/regolith-desktop#742] [Ideas]
Hey!

I've been with Regolith now since 1.1, and run a mix of versions; 1.6 on an older work thinkpad which I am soon killing off, and 2.X on a framework laptop

Overall, I love the experience. Using the new partials setup in Regolith 2 took some getting used to, but now I love it. Having my small configs for specific tweaks I have made my test of 'wiping and re-configuring' a laptop very smooth

However, the framework I'm using as well as newer external monitors at my work, are all rocking HiDPI screens now.. and Xorg is starting to show its age real hard. I know there's an [older issue](https://github.com/regolith-linux/regolith-desktop/issues/27) around a migration, but I didn't see it here on the GH discussions. So I thought I'd bring it up.

For what it's worth, I found this [little repo](https://github.com/regolith-linux/regolith-sway) which shows some external attempt has been made.

Otherwise then, what's the opinion of others and you, @kgilmer, on the future of regolith and possible Sway compat? What are the blockers? (I know the heavy usage of Xresouces is one such item). I'd like to try and help with this, if possible.

**Comment by kgilmer** [comment]
Hi @Jarvvski , 

  Thanks for the feedback on Regolith in general, and happy to hear you (eventually) came around to using partials to manage i3.  Regarding Wayland (via Sway) support, in fact there has been a lot of work going into this lately, almost all of it by @SoumyaRanjanPatnaik .  You can check out [his GSoC post](https://regolith-desktop.com/posts/sway_integration_gsoc/) to get a good detailed overview of what's done and where we currently are in development.  We're tracking deliverables in [this project](https://github.com/orgs/regolith-linux/projects/8).  Let us know if you'd like to dig in as well, plenty to do!

**Comment by Jarvvski** [comment]
@kgilmer awesome! That's great to hear

I did take a look see at the project board. Is there anyway to highlight potentially 'good first issue' style cards? I think GH project items don't allow labels the same way issues do..

But I'd love to help out. I just don't know where's a good place to get started currently 😄

**Comment by kgilmer** [comment]
Presently I'm focused on an upcoming release to support the latest Ubuntu release.  Once that's out the door I'll hook back up with @SoumyaRanjanPatnaik to help out where I can.  I believe most of the code is either Rust or Bash.  @SoumyaRanjanPatnaik , any suggestions for starter tasks on the Wayland project?

**Reply by SoumyaRanjanPatnaik** [reply]
I can think of a few tasks for `regolith-inputd`. We also have to work on packaging for a lot of repositories. Once we come up with a structure for sway configurations, there should also be the task of reorganising binary packages and configuration files in the `regolith-i3-config` package. I actually haven't been able to work properly for the past two weeks to meet my academic requirements. So any contributions are definitely welcome.
